### DISCUSSION TITLE
2 locations of recording posible?

### DISCUSSION THREAD

**Discussion by JA16122000** [mrlt8/docker-wyze-bridge#1259] [Q&A]
Is it possible to save recordings to two different locations?

Locally on the Raspberry Pi and on a network path.

**Comment by mrlt8** [comment]
Could you use an NVR or mount the network storage to the container?

FYI recording will be changing in the next version see #1224.

**Reply by JA16122000** [reply]
I have a network drive connected to my FritzBox 5590 modem, but I will wait for the upcoming change before proceeding with dual recordings.

When will this change be implemented?

**Reply by mrlt8** [reply]
Trying to get the current version stable before merging the changes, but you can test out the new version with the `dev` image.

**Reply by JA16122000** [reply]
🚀 DOCKER-WYZE-BRIDGE v2.9.7 ARMV7L DEV BUILD [2024-06-11t14:21:24.631z] 9842374
[WyzeBridge] ♻️ Clearing local cache...
[WyzeBridge] [+] Custom stream auth enabled
[WyzeBridge] [MTX] Auth [mulderij:7316kp04] paths='all'
[WyzeBridge] 🔍 Could not find local cache for 'auth'
[WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
 * Serving Flask app 'frontend'
 * Debug mode: off
[WyzeBridge] WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://172.18.0.2:5000
[WyzeBridge] Press CTRL+C to quit
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:00] "GET /?video HTTP/1.1" 302 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:00] "GET /login HTTP/1.1" 200 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:00] "GET /static/bulma-toast.js HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:00] "GET /static/site.css HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:00] "GET /static/bulma.css HTTP/1.1" 304 -
[WyzeBridge] 💾 Saving 'auth' to local cache...
[WyzeBridge] 🔍 Could not find local cache for 'user'
[WyzeBridge] ☁️ Fetching 'user' from the Wyze API...
[WyzeBridge] 💾 Saving 'user' to local cache...
[WyzeBridge] 🔍 Could not find local cache for 'cameras'
[WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:02] "GET /login HTTP/1.1" 302 -
[WyzeBridge] 🔍 Could not find local cache for 'cameras'
[WyzeBridge] ☁️ Fetching 'cameras' from the Wyze API...
[WyzeBridge] [API] Fetched [2] cameras
[WyzeBridge] 💾 Saving 'cameras' to local cache...
[WyzeBridge] [+] Adding Voortuin [HL_CAM3P]
[WyzeBridge] [API] Fetched [2] cameras
[WyzeBridge] 💾 Saving 'cameras' to local cache...
[WyzeBridge] 📹 Will record 1800ss clips to %/record/{cam_name}
[WyzeBridge] [+] Adding Achtertuin [HL_CAM3P]
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:04] "GET / HTTP/1.1" 200 -
[WyzeBridge] 📹 Will record 1800ss clips to %/record/{cam_name}
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:04] "GET /static/bulma.css HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:04] "GET /static/site.css HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:04] "GET /static/bulma-toast.js HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:04] "GET /static/site.js HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:04] "GET /static/webrtc.js HTTP/1.1" 304 -
[WyzeBridge] starting MediaMTX 1.8.2
[WyzeBridge] 🎬 2 streams enabled
RtspEvent initialized
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:04] "GET /api/sse_status HTTP/1.1" 200 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:11] "GET / HTTP/1.1" 200 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:11] "GET /static/site.js HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:11] "GET /static/bulma.css HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:11] "GET /static/site.css HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:11] "GET /static/bulma-toast.js HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:11] "GET /static/webrtc.js HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:11] "GET /static/loading.svg HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:11] "GET /api/sse_status HTTP/1.1" 200 -
[WyzeBridge] 🎉 Connecting to WyzeCam V3 Pro - Achtertuin on 192.168.178.69
[WyzeBridge] 🎉 Connecting to WyzeCam V3 Pro - Voortuin on 192.168.178.70
[WyzeBridge] 🎉 Connecting to WyzeCam V3 Pro - Voortuin on 192.168.178.70
[WyzeBridge] 🎉 Connecting to WyzeCam V3 Pro - Achtertuin on 192.168.178.69
[WyzeBridge] ☁️ Pulling "achtertuin" thumbnail to /img/achtertuin.jpg
[WyzeBridge] ☁️ Pulling "voortuin" thumbnail to /img/voortuin.jpg
Connection closed by remote. Closing connection.
Connection closed by remote. Closing connection.
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:13] "GET /img/voortuin.jpg HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:13] "GET /img/achtertuin.jpg HTTP/1.1" 304 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:13] "GET /img/voortuin.jpg HTTP/1.1" 304 -
[voortuin] [-22] IOTC_ER_SESSION_CLOSE_BY_REMOTE
Process voortuin:
Traceback (most recent call last):
  File "/app/frontend.py", line 202, in img
    return send_from_directory(config.IMG_PATH, img_file)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/flask/helpers.py", line 552, in send_from_directory
    return werkzeug.utils.send_from_directory(  # type: ignore[return-value]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/werkzeug/utils.py", line 575, in send_from_directory
    raise NotFound()
werkzeug.exceptions.NotFound: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/app/wyzebridge/wyze_stream.py", line 448, in start_tutk_stream
    if ffmpeg and ffmpeg.poll() is None:
       ^^^^^^
UnboundLocalError: cannot access local variable 'ffmpeg' where it is not associated with a value
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:14] "GET /img/achtertuin.jpg HTTP/1.1" 304 -
[voortuin] 📡 Getting 180kb/s 2K stream (H264/20fps) via LAN mode (WiFi: 59%) FW: 4.58.12.9751 🔒
[achtertuin] 📡 Getting 180kb/s 2K stream (H264/20fps) via LAN mode (WiFi: 37%) FW: 4.58.12.9751 🔒
[voortuin] WARNING: Skipping wrong frame_size at start of stream [frame_size=1]
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:14] "GET /signaling/voortuin?webrtc HTTP/1.1" 200 -
[WyzeBridge] 195.240.59.44 - - [18/Jun/2024 17:03:14] "GET /signaling/achtertuin?webrtc HTTP/1.1" 200 -
[WyzeBridge] ✅ '/voortuin stream is UP! (3/3)
[WyzeBridge] 📖 New client reading from voortuin
[achtertuin] WARNING: Skipping wrong frame_size at start of stream [frame_size=1]
[WyzeBridge] ✅ '/achtertuin stream is UP! (3/3)
[WyzeBridge] 📖 New client reading from achtertuin

**Reply by JA16122000** [reply]
docker-compose:

services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: unless-stopped
        image: mrlt8/wyze-bridge:dev
        ports:
            - 1935:1935 # RTMP
            - 8554:8554 # RTSP
            - 8888:8888 # HLS
            - 5000:5000 # WEB-UI
            - 9997:9997 # Expose port for API
            - 8889:8889 # WebRTC
            - 8189:8189/udp # WebRTC/ICE
        environment:
            - API_ID=blob
            - API_KEY=blob
            - WYZE_EMAIL=blob
            - WYZE_PASSWORD=blob
            - STREAM_AUTH=blob:blob
            - WB_AUTH=True
            - WB_USERNAME=blob
            - WB_PASSWORD=blob
            - WB_IP=192.168.178.78
            - NET_MODE=LAN
            - KEEP_BAD_FRAMES=True
            - FRESH_DATA=True
            - TZ=Europe/Amsterdam
            - RECORD_ALL=True
            - RECORD_PATH=%/Record/{cam_name}
            - RECORD_FILE_NAME={cam_name}_%Y_%m_%d_%H_%M_%S%f%s
            - RECORD_LENGTH=1800s    
        volumes:
            - /mnt/usbhdd/Record/:/record/
            - ./tokens:/tokens/

**Reply by JA16122000** [reply]
There is absolute NO recording. [ffmpeg related?]

I paste the log in bard.google.com and this is what it answered:

==

It seems that WyzeBridge is successfully connecting to your Wyze Cam V3 Pro cameras ('Front Yard' and 'Back Yard'), but there are a couple of minor errors.

Let's break down the log:

    Initially, everything seems to be going well. WyzeBridge starts up and fetches the necessary information from the Wyze API.
    Then, both cameras are detected and WyzeBridge attempts to connect.
    The connection is successful and WyzeBridge can even retrieve stream information (kb/s, fps, etc.).
    After that, WyzeBridge tries to download a thumbnail image (/img/front_yard.jpg and /img/back_yard.jpg) but gets a 404 Not Found error. This could mean that the thumbnails are not yet available.

Errors:

    UnboundLocalError: This error message appears for the 'front_yard' camera. It seems that a variable named 'ffmpeg' is not defined correctly.
    WARNING: Skipping wrong frame_size at start of stream: This warning appears for both cameras. It could indicate a minor issue with the video stream at the beginning, but it seems to recover afterward.

Everything else seems to be working:

Despite the minor errors, WyzeBridge seems to be able to connect to your cameras and even display video streams.

Possible solutions:

    Thumbnail error: This error is not critical, but it can be annoying. It could be that the thumbnails are not yet available. You can try restarting WyzeBridge or waiting for the thumbnails to be generated.
    UnboundLocalError: This error message requires further investigation. There might be a bug in the WyzeBridge software. You can try installing the latest version or searching for information about this specific error message.

**Reply by mrlt8** [reply]
Your `RECORD_PATH` was wrong as mediamtx uses `%path` as the stream name and doesn't use `RECORD_FILE_NAME`.

However, I made some changes so that the dev branch should work with the old formatting from the previous versions of the bridge.

**Reply by JA16122000** [reply]
Still no recording with:

===================

services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: unless-stopped
        image: mrlt8/wyze-bridge:dev
        ports:
            - 1935:1935 # RTMP
            - 8554:8554 # RTSP
            - 8888:8888 # HLS
            - 5000:5000 # WEB-UI
            - 9997:9997 # Expose port for API
            - 8889:8889 # WebRTC
            - 8189:8189/udp # WebRTC/ICE
        environment:
            - API_ID=blob
            - API_KEY=blob
            - WYZE_EMAIL=blob
            - WYZE_PASSWORD=blob
            - STREAM_AUTH=blob:blob
            - WB_AUTH=True
            - WB_USERNAME=blob
            - WB_PASSWORD=blob
            - WB_IP=192.168.178.78
            - NET_MODE=LAN
            - KEEP_BAD_FRAMES=True
            - FRESH_DATA=True
            - TZ=Europe/Amsterdam
            - RECORD_ALL=True
            - RECORD_PATH=/Record/{cam_name}
            - RECORD_FILE_NAME={cam_name}_%Y_%m_%d_%H_%M_%S
            - RECORD_LENGTH=1800s    
        volumes:
            - /mnt/usbhdd/Record/:/record/
            - ./tokens:/tokens/

**Reply by mrlt8** [reply]
Your record path is `Record`
```yaml
- RECORD_PATH=/Record/{cam_name}
```
But you're mounting `record`
```yaml 
- /mnt/usbhdd/Record/:/record/
```

**Reply by JA16122000** [reply]
Changed '/record/' to '/Record/' and now it is recording ;) [On the latest and previous build, this won't record at all.]
But now i don't get a live stream.
Just an frozen image

HLS Stream is working fine, but WebRTC stream is not working

**Reply by mrlt8** [reply]
Please report any bugs related to the dev branch in #1224. Thanks.

**Reply by JA16122000** [reply]
> Changed '/record/' to '/Record/' and now it is recording ;) [On the latest and previous build, this won't record at all.] But now i don't get a live stream. Just an frozen image
> 
> HLS Stream is working fine, but WebRTC stream is not working

@mrlt8 I can not reply anymore on: https://github.com/mrlt8/docker-wyze-bridge/pull/1224
But i've noticed when i put this in my environment, there is NO problem enymore with VPN use.

            - WB_IP=192.168.178.78    <== Internal IP from the bridge
            - WB_IP=195.240.59.44     <== External ISP adress (Just added for testing)
            

With external ISP adress added, there is no VPN issue like in the other post

**Comment by JA16122000** [comment]
So in mine opinion, whe have to put 

               - WB_IP=[HERE YOUR EXTERNAL IP] 
               
Then there should no problem enymore
